### DISCUSSION TITLE
Universe 2024 Workshop: Continuous Deployments the Github Way

### DISCUSSION THREAD

**Discussion by igwejk** [community/community#143136] [Other Feature Feedback, Questions, & Ideas]
## Universe 2024 Workshop: Continuous Deployments the Github Way

Although continuous deployment is a well-practiced concept in the industry, it appears many organization are still faced with the challenge of realising an effective and simplified implementation. This workshop, _"Continuous deployments, the GitHub way"_ demonstrates an archetypal implementation that gives due consideration to security and compliance; and achieves simplicity by identifying _[bounded contexts](https://www.dremio.com/wiki/bounded-context/)_ in the CI/CD domain that are fitting to relatable business value streams and mapping respective contexts to appropriate GitHub events.

Essentially, the workshop is self-paced and can be can be independently completed anytime. During the live GitHub Universe session however, you will have the opportunity to directly interact with a GitHub expert and a super-user enterprise customer who already implemented the concepts with a successful outcome.

This workshop assumes participants already have some knowledge of GitHub Actions. Nevertheless, it is possible to follow and understand the demonstrations without prior experience of GitHub Actions or participants may refer to [**GitHub DevSecOps Fundamentals**](https://gh-devsecops.github.io/fundamentals) self-paced workshop to address basic gaps.

## What you will learn

- [x] Apply event-driven system design thinking to continuous deployment
- [x] Ensure compliance provenance and attestation for build artifacts
- [x] Harden security of integration with a target cloud environment, using Open-ID Connect for implementation of identity federation

## Speakers
* Igwe Kalu (@igwejk), Sr. Service Delivery Engineer, GitHub Social
  [`X` (igwejk)](https://twitter.com/igwejk) / [`LinkedIn` (igwejk)](https://www.linkedin.com/in/igwejk)
* Turgay Akbas (@akbast), DevOps / Platform Engineer compeople AG

> [!IMPORTANT]
> ## Take home resources
> 
> [**Continuous Deployments the Github Way, workshop**](https://demos-by-igwejk.github.io/continuous-deployments-the-github-way/).

**Comment by regicsolutions** [comment]
@igwejk Thanks for starting this discussion. You mentioned something around claims during the demo are you able to provide some more information on that?

**Reply by igwejk** [reply]
Hi @regicsolutions, I'm glad you've followed up with this valuable question. [**Defining the trust conditions**](https://docs.github.com/en/enterprise-cloud@latest/actions/security-for-github-actions/security-hardening-your-deployments/using-openid-connect-with-reusable-workflows#defining-the-trust-conditions), addresses that point.
